git_machete/client.py

```diff
@@ -1512,8 +1512,13 @@ def __pick_remote(self, branch: str, is_called_from_traverse: bool) -> None:
         print("\n".join(f"[{index + 1}] {rem}" for index, rem in enumerate(rems)))
         msg = f"Select number 1..{len(rems)} to specify the destination remote " \
               "repository, or 'n' to skip this branch, or " \
-              "'q' to quit the traverse: "
+              "'q' to quit the traverse: " if is_called_from_traverse \
+              else f"Select number 1..{len(rems)} to specify the destination remote " \
+                   "repository, or 'q' to quit creating pull request: "
+
         ans = input(msg).lower()
+        if not is_called_from_traverse and not ans.isnumeric():
+            raise MacheteException('Could not establish remote repository, pull request creation interrupted.')
         if ans in ('q', 'quit'):
```

Comment:
Nit: maybe swap this `if` and the preceding one? this way `q` is handled quietly, without somewhat redundant/unexpected `Could not establish remote repository...` message

Reply:
Although then `StopTraversal` should probably be renamed to sth more adequate since it longer serves just for the purpose of well, stopping the traversal